File: externals/Async-CTP/samples/CS/Windows/Phone/Concert/Finder/Concert/Finder/Converters/TimeConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace ConcertFinder.Converters
{
    /// <summary>
    /// Converter for DateTime values.
    /// </summary>
    public class TimeConverter : IValueConverter
    {
        /// <summary>
        /// Convert a DateTime value to a string using the given format string parameter.
        /// </summary>
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var convertedValue = String.Empty;

            var originalValue = value as DateTime?;
            var format = parameter as string;
            if (originalValue.HasValue && !String.IsNullOrEmpty(format))
            {
                convertedValue = originalValue.Value.ToString(format);
            }

            return convertedValue;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: externals/Async-CTP/samples/VB/Windows/Phone/Concert/Finder/Concert/Finder/Converters/TimeConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows.Data

Namespace ConcertFinder.Converters

    ''' <summary>
    ''' Converter for DateTime values.
    ''' </summary>
    Public Class TimeConverter
        Implements IValueConverter

        ''' <summary>
        ''' Convert a DateTime value to a string using the given format string parameter.
        ''' </summary>
        Public Function Convert(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim convertedValue = [String].Empty
            Dim originalValue = CType(value, DateTime?)
            Dim format = TryCast(parameter, String)
            If originalValue.HasValue AndAlso Not [String].IsNullOrEmpty(format) Then
                convertedValue = originalValue.Value.ToString(format)
            End If
            Return convertedValue
        End Function

        Public Function ConvertBack(value As Object, targetType As [Type], parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

    End Class

End Namespace